### PR TITLE
chore: updated SDK versions/snippets

### PR DESCRIPTION
**Pull request by falconandy**
# Amplitude Developer Docs PR

## Description

Updated SDK versions/snippets

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [x] Doc update.
- [ ] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [x] My documentation follows the style guidelines of this project.
- [x] I previewed my documentation on a local server using `mkdocs serve`.
- [x] Running `mkdocs serve` didn't generate any failures.
- [x] I have performed a self-review of my own documentation.

@amplitude-dev-docs
@caseyamp

### CODE REVIEW

**Comment by caseyamp** [comment]
@falconandy you can ignore the link check warnings

**Review by justin-fiedler — CHANGES_REQUESTED** [review]

File: docs/data/sdks/java/ampli.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ Amplitude Data supports tracking analytics events from JRE programs written in J
 1. [Install the Amplitude SDK](#install-the-amplitude-sdk)
 
     ```java
-    implementation 'com.amplitude:java-sdk:1.10.0'
+    implementation 'com.amplitude:java-sdk:1.+'
```
[the remark this answers is not in the document]
Comment (falconandy) [reply docs/data/sdks/java/ampli.md:29]: I think, versions in the documentation are used by new users mostly. `1.+` version will still install the latest available version for them. 
Maybe documentation sections (like `Middleware support`) should have some badges with supported version - like `Available from version 1.6`

[the remark this answers is not in the document]
Comment (falconandy) [reply docs/data/sdks/java/ampli.md:29]: Updated based on https://github.com/amplitude/javascript/blob/master/server/packages/orbit/apps/stargate-client/src/codegen/sdk-versions.ts

File: docs/data/sdks/java/ampli.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ Amplitude Data supports tracking analytics events from JRE programs written in J
 1. [Install the Amplitude SDK](#install-the-amplitude-sdk)
 
     ```java
-    implementation 'com.amplitude:java-sdk:1.+'
+    implementation 'com.amplitude:java-sdk:[1.8.0,2.0)'
```
Comment (justin-fiedler) [inline docs/data/sdks/java/ampli.md:29]: I think this needs to be `[1.10.0,2)` for Plan information.

**Review by justin-fiedler — APPROVED** [review]

File: docs/data/sdks/java/ampli.md (hunk 2 above)
Comment (justin-fiedler) [reply docs/data/sdks/java/ampli.md:29]: Nevermind, plan is in 1.8.0

**Pull request merged**